packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -303,29 +303,96 @@ class DateTimePicker extends React.Component {
+
+    if (event.type === 'keydown') {
+      // Keyboard shortcut is pressed, split the time value into hour and minute inputs
+      const timeArray = this.timeValue.split(':');
+      if (event.key === 'n' || event.key === 'N') {
+        // `N` sets Time to now, so Date should be set to today
+        previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+        validDate = true;
+        dateChanged = true;
+      } else if (event.key === '-' || event.key === '_') {
+        // `-` subtracts 1 minute`from time, so check date to see if it needs changed
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          validDate = true;
+          dateChanged = true;
+        }
+
+        if (timeArray[0] === '23' && timeArray[1] === '59') {
+          // Time is 23:59 which means previous datetime should be on the wrong day, lower it by 1
+          previousDateTime = previousDateTime.subtract(1, 'd');
+          dateChanged = true;
+        }
+
+        const changedDateTime = DateTimeUtils.updateTime(previousDateTime, this.timeValue, this.props.showSeconds);
+        const checkHour = previousDateTime.clone().subtract(1, 'minute').hours();
+        if (changedDateTime.isDST() && checkHour !== changedDateTime.hours()) {
+          // Datetime falls under DST and hour is wrong using moment logic, so rewrite it using moment logic
+          timeArray[0] = (`0${checkHour}`).slice(-2);
+          this.timeValue = timeArray.join(':');
+        }
+      } else if (event.key === '=' || event.key === '+') {
+        // `+` add 1 minute to time, so check date to see if it needs changed
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          validDate = true;
+          dateChanged = true;
+        }
+
+        if (timeArray[0] === '00' && timeArray[1] === '00') {
+          // Time is 00:00 which means previous datetime should be on the wrong day, increase it by 1
+          previousDateTime = previousDateTime.add(1, 'd');
+          dateChanged = true;
+        }
+      } else if (this.props.timeVariant === DateTimeUtils.FORMAT_12_HOUR && (event.key === 'a' || event.key === 'A' || event.key === 'p' || event.key === 'P')) {
+        // `A` and `P`, changes the meridiem, so check date to see if it needs changed
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          validDate = true;
+          dateChanged = true;
+        }
+      }
+    }
+    if (dateChanged) {
+      this.handleDateChange(event, previousDateTime.format('YYYY-MM-DD'), null, true);
```

Comment:
I thought this was removed based on this [comment](https://github.com/cerner/terra-framework/pull/1520#discussion_r815213080). What is the reason for adding this back as this will trigger the onChange callback twice?

Reply:
After a closer look, I see what is being done here. It feels hacky and anti-pattern to do things this way by calling handeDateChange. If it needs any state or class variables need to be updated, it should be more direct to just update what is needed directly here instead of relying and cherry picking the logic in handleDateChange.

Reply:
Fixed Here https://github.com/cerner/terra-framework/pull/1520/commits/3ad7a15a91e5339a735c4c70ca93065c8d227b64#diff-063daa4349be2bbad9c58209d280bfaee7a04b7e429163ebdb7784768f501a5cL388